### DISCUSSION TITLE
Ways to mitigate a crashing clangd? (I think it filled my disk)

### DISCUSSION THREAD

**Discussion by rnikander** [joaotavora/eglot#1075] [Q&A]
I've got a C++ project, and I started using eglot. It was going well for a few days, but now sometimes everything gets into this unusable state where I keep seeing the message in the minibuffer: `[eglot] Connected! Server ``clangd' now managing ``(c++-mode c-mode)' buffers in project ``MyApp'`.  And the CPU fans start making noise, Emacs performance goes down the drain. 

I investigated a bit and I think clang is crashing. I tried looking at this `*EGLOT... events*` buffer. It's hard to follow because it's constantly getting rewritten and the cursor is jumping around, but I can select all the text and copy it out of there, and then see the seg fault stack trace from clang:

```text
[stderr] Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
[stderr] 0  clangd                   0x000000010ff09c3e llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 46
[stderr] 1  clangd                   0x000000010ff08ba6 llvm::sys::RunSignalHandlers() + 198
[stderr] 2  clangd                   0x000000010ff0a310 SignalHandler(int) + 272
[stderr] 3  libsystem_platform.dylib 0x00007ff802cf9c1d _sigtramp + 29
[stderr] 4  libsystem_platform.dylib 0xfffffffffffff000 _sigtramp + 18446603370533770240
[stderr] 5  clangd                   0x00000001100a0dd3 EvaluateInPlace(clang::APValue&, (anonymous 
...
[stderr] 24 libsystem_pthread.dylib  0x00007ff802ccac7b thread_start + 15
[stderr] Signalled during AST worker action: Build AST
[stderr]   Filename: .../BlockElt.cc
[stderr]   Directory: .../MyApp/build-debug
[stderr]   Command Line: /Users/rob/Dev/clang+llvm-15.0.1-x86_64-apple-darwin/bin/clang++ [....]
```

So I was going to ask: is there a good way to tell eglot to stop trying? Then I can re-enable it after a few minutes when I've edited the files some. Maybe `eglot-shutdown`? I tried that but I thought it tried to restart as I visited files. But Emacs was going to hell so it was hard to tell what was going on.

As I was writing this, it was crashing in the background, and I just got a message from macOS that my disk was almost full! I don't think I was anywhere near full (500GB) so I need to investigate now to see if something was writing a giant log file, and where it might have put it.  If you have any idea about that, please let me know.  It's not in my CMake build directory.

**Comment by joaotavora** [comment]
> I tried that but I thought it tried to restart as I visited files.

Perhaps you're using `eglot-ensure`, which i don't really recommend unless you are confident that the server can be started smoothly for any given project. Try with `M-x eglot` instead and then you can probably control things with `eglot-shutdown` better. Also see the manual's troubleshooting section.

**Reply by rnikander** [reply]
Okay, thanks.

I think I found the issue with drive space too. There were about 100 GB of what look like temporary files under /var/folders, with names matching `preamble-*.pch` and some others.
